src/main/utils/resource.ts

```diff
@@ -1,112 +1,112 @@
-//     private static readonly MILLISECONDS_IN_HOUR: number = 3600000;
-//     private TMP_DOWNLOAD_PATH: string = '';
-//     private resourcePath: string = '';
-//     private readonly _defaultJfrogClient: JfrogClient = ConnectionUtils.createJfrogClient(
-//         'https://releases.jfrog.io',
-//         'https://releases.jfrog.io/artifactory',
-//         '',
-//         '',
-//         '',
-//         ''
-//     );
+// Resource classes represent generic instances of external artifact (e.g. binary).
+export class Resource {
+    // You should not change this variable. This may only be done for testing purposes.
+    public static MILLISECONDS_IN_HOUR: number = 3600000;
+    // From which to download from.
+    private downloadTarget: string;
+    // To which dir download the resource.
+    private downloadDir: string;
+    //A local path to the resource
+    private path: string;
+    // Resource's sha256
+    private sha2: string | undefined;
 
-//     constructor(
-//         private resourceLocalPath: string,
-//         private resourceRemotePath: string,
-//         private resourceName: string,
-//         private _logManager: LogManager,
-//         private _connectionManager?: ConnectionManager
-//     ) {
-//         this.TMP_DOWNLOAD_PATH = path.join(resourceLocalPath, 'download');
-//         this.resourcePath = path.join(resourceLocalPath, resourceName);
-//     }
+    private _connectionManager: JfrogClient = ConnectionUtils.createJfrogClient(
+        'https://releases.jfrog.io',
+        'https://releases.jfrog.io/artifactory',
+        '',
+        '',
+        '',
+        ''
+    );
 
-//     public async download() {
-//         if (!fs.existsSync(this.TMP_DOWNLOAD_PATH)) {
-//             fs.mkdirSync(this.TMP_DOWNLOAD_PATH, { recursive: true });
-//         } else if (Date.now() - fs.statSync(this.TMP_DOWNLOAD_PATH).birthtimeMs <= Resource.MILLISECONDS_IN_HOUR) {
-//             // By here, someone else is already downloading the scanner.
-//             return;
-//         } else {
-//             // Seems like it is a left over from other download.
-//             fs.rmSync(this.TMP_DOWNLOAD_PATH);
-//         }
-//         try {
-//             if (this._connectionManager !== undefined) {
-//                 await this._connectionManager.downloadArtifactToFile(this.resourceRemotePath, path.join(this.TMP_DOWNLOAD_PATH, this.resourceName));
-//             } else {
-//                 await this._defaultJfrogClient
-//                     .artifactory()
-//                     .download()
-//                     .downloadArtifactToFile(this.resourceRemotePath, path.join(this.TMP_DOWNLOAD_PATH, this.resourceName));
-//             }
+    constructor(
+        private _homeDir: string,
+        private downloadSource: string,
+        resourceName: string,
+        private _logManager: LogManager,
+        connectionManager?: JfrogClient
+    ) {
+        this.downloadDir = path.join(_homeDir, 'download');
```

Comment:
I'm not sure that using a **temporary** dir under the user home is a good idea. The system's temp folder is meant for this.
Also - if the user closes his vscode during download, the next time he opens vscode may cause freeze for an hour. Using a temp folder in the system temp path should resolve this.